src/NServiceBus.Core/Transports/Learning/DirectoryBasedTransaction.cs

```diff
@@ -0,0 +1,101 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+
+    class DirectoryBasedTransaction : ILearningTransportTransaction
```

Comment:
We have no exception handling here. So we assume things to bomb and bubble up if anything happens in between operations for example during commit?